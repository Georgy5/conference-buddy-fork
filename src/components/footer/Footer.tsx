import React from 'react'

function Footer() {
  return (
    <footer>
      <div className="bg-light py-2 small text-center shadow-lg">
        ü§ê
        <a
          className="TextLink TextLink-primary"
          href="https://www.iubenda.com/privacy-policy/44138766"
        >
          Privacy Policy
        </a>
      </div>
      <div className="container-fluid small bg-dark text-light text-center">
        Made with ‚ù§Ô∏è lots of ‚òïÔ∏è and an awesome ‚å®Ô∏è
      </div>
    </footer>
  )
}

export default Footer
