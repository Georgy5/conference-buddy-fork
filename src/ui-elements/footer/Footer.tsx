import React, { ReactElement } from "react"
import { TextLink } from "../text-link/TextLink"

function Footer(): ReactElement {
  return (
    <div className="bg-dark pt-4 pt-md-5 pb-3">
      <div className="container text-white pb-2">
        <div className="row justify-content-evenly">
          <div className="col-12 col-md-3 text-center text-center">
            <h5>Platform</h5>
            <nav>
              <ul className="list-unstyled">
                <li>
                  <TextLink internal={true} light={true} to={"/"}>
                    How does this work?
                  </TextLink>
                </li>
                <li>
                  <TextLink internal={true} light={true} to={"/"}>
                    Support ConfBuddy
                  </TextLink>
                </li>
              </ul>
            </nav>
          </div>

          <div className="col-12 col-md-3 text-center mt-3 mt-md-0">
            <h5>Social</h5>
            <nav>
              <ul className="list-unstyled">
                <li>
                  <TextLink
                    internal={false}
                    light={true}
                    to="https://www.iubenda.com/privacy-policy/44138766"
                  >
                    Twitter
                  </TextLink>
                </li>
                <li>
                  <TextLink
                    internal={false}
                    light={true}
                    to="https://www.iubenda.com/privacy-policy/44138766"
                  >
                    Github
                  </TextLink>
                </li>
              </ul>
            </nav>
          </div>

          <div className="col-12 col-md-3 text-center mt-3 mt-md-0">
            <h5>About</h5>
            <nav>
              <ul className="list-unstyled">
                <li>
                  <TextLink internal={false} light={true} to="">
                    Privacy Policy
                  </TextLink>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </div>
      <footer>
        <div className="container small bg-dark text-light text-center">
          Made with ❤️ lots of ☕️ and an awesome ⌨️
        </div>
      </footer>
    </div>
  )
}

export { Footer }
