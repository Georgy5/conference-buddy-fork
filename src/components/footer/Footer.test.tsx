import { render, screen, cleanup, RenderResult } from "@testing-library/react"
import Footer from "./Footer"

describe("Footer", () => {
  let component: RenderResult
  beforeAll(() => {
    component = render(<Footer />)
  })

  afterAll(cleanup)

  it("renders a footer element", () => {
    const footer = screen.getByRole("contentinfo")
    expect(footer).toBeVisible()
  })

  it("renders a link to privacy policy", () => {
    const link = screen.getByRole("link", {
      name: "Privacy Policy",
    })
    expect(link).toBeVisible()
  })

  it("renders all necessary elements", () => {
    expect(component.asFragment()).toMatchSnapshot()
  })
})
