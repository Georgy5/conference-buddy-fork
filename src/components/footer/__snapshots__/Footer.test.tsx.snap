// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders all necessary elements 1`] = `
<DocumentFragment>
  <footer>
    <div
      class="bg-light py-2 small text-center shadow-lg"
    >
      ü§ê
      <a
        class="TextLink TextLink-primary"
        href="https://www.iubenda.com/privacy-policy/44138766"
      >
        Privacy Policy
      </a>
    </div>
    <div
      class="container-fluid small bg-dark text-light text-center"
    >
      Made with ‚ù§Ô∏è lots of ‚òïÔ∏è and an awesome ‚å®Ô∏è
    </div>
  </footer>
</DocumentFragment>
`;
